plasmapy/particles/particle_class.py

```diff
@@ -2061,3 +2064,9 @@ def charge(self, q: Optional[Union[u.Quantity, Real]]):
                 "a real number that represents the ratio of the charge to "
                 "the elementary charge."
             )
+
+
+# TODO: Describe valid particle representations in docstring of particle_like
+
+particle_like = Union[str, Integral, Particle]
+"""A typing construct for valid representations of a particle."""
```

Comment:
@rocco8773 — With respect to [your comment](https://github.com/PlasmaPy/PlasmaPy/issues/912#issuecomment-706764253) in #912, I tentatively added a stub for `particle_like` here.  I started using it in this PR, so it'd be easiest for me if we could finish this PR, and then work on a separate PR that implements your ideas and puts `particle_like` in a more complete form, possibly in a different location.

Also:
 - `numbers.Integral` corresponds to the base class for integers representing atomic numbers.  
 - I used all lower case in `particle_like` since types like `int` and `float` are typically lower case, and because I think it's an instance rather than an actual class.  (With that said, I don't really have a good idea of the inner workings of `typing`.)

Reply:
That's fine.  We can discuss `particle_like` vs `ParticleLike` in #912 and the subsequent PR.  I did choose `ParticleLike` because that's the nomenclature used in [Python's documentation for typing aliases](https://docs.python.org/3/library/typing.html#type-aliases).  (Even though I personally prefer `particle_like`).

With regards to `number.Integral`, it is not a traditional base class for for `int` (i.e. `int` does not inherit from it...`int.__mro__ == (int, object)`).  Instead it is an [abc "registered" subclass](https://docs.python.org/3/library/abc.html#abc.ABCMeta.register), or a "virtual subclass".